packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -496,16 +498,9 @@ If you provided a React element for the optionText prop, you must also provide t
                 onChange={handleAutocompleteChange}
                 onBlur={field.onBlur}
                 onInputChange={handleInputChange}
-                renderOption={(props, record) => {
-                    if (isValidElement(optionText)) {
```

Comment:
Why did you remove this one? This removes the ability to pass an element as optionText

Reply:
Nope, see `getChoiceText`